### ISSUE TITLE
Interaction with Excalidraw

### ISSUE THREAD

**Issue by LeonelRFF** [logancyang/obsidian-copilot#514]
**Is your feature request related to a problem? Please describe.**
Hi, I can't get it to interact with Excalidraw

**Describe the solution you'd like**
Would it be possible to integrate it with the Excalidraw plugin (the most downloaded plugin so far)? Currently many of us use it for sketching

**Describe alternatives you've considered**
In Copilot, custom templates could be added to indicate that in certain types of files it only reads the text that is under a specific title. In the case of Excalidraw, it could be that it reads under the title Excalidraw data/Text Elements as can be seen in the video

https://github.com/user-attachments/assets/5d0fbea2-15d0-403a-a036-c4e870d9171d

**Comment by logancyang** [comment]
Interesting, what does the raw `.excalidraw` file look like in this case? You can turn on debug mode and check the dev console, it will show the actual full message sent to the LLM. Or you can check your Ollama server log since you are using Ollama in this case.

**Comment by LeonelRFF** [comment]
Try testing with another example markdown note, age of empires for this case, and if it works, but you can't do it in excalidraw


https://github.com/user-attachments/assets/d2022a6e-6496-48cd-b270-86984bdc045f


https://github.com/user-attachments/assets/2ebeab03-69d8-407e-8c07-2ece22d948db

I think it looks like everything is apparently working fine, maybe the context window is not enough since in addition to that text in excalidraw another section called Drawing, there it contains information that is not necessary, that is why my suggestion is that in certain types of files specify where it should read


https://github.com/user-attachments/assets/cb1bf0c3-f182-4345-81f1-1fa3cfd992e6

obviously the videos are accelerated so that they can be attached here

**Comment by zitongcharliedeng** [comment]
+1 And the ability to know what current items are in-focus and for the diff viewer to be visual for excalidraw.
This'll make other AI canvas drawing apps redundant and keep me in my vault for good!

Many, MANY people use Zsolt's visual note taking method (4D PKD) - he'd probably be willing to help if he doesn't end up making it native as an excalidraw agent extension himself.

I myself no longer write ANY markdown and write all ideal state artifacts purely as linked excalidraw blueprints.

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
